### ISSUE TITLE
'WiFi' was not declared in this scope

### ISSUE THREAD

**Issue by RobertSchuster** [CircuitSetup/Split-Single-Phase-Energy-Meter#81]
Compiling with Arduino IDE failes with _**'WiFi' was not declared in this scope**_

See attached file.
[compile Error.txt](https://github.com/user-attachments/files/18252511/compile.Error.txt)

Arduino IDE and all libraries are up2date.
Any ideas?

**Comment by ptownhippie** [comment]
I read somewhere about changing something in the file concerning HTTP. Somewhere in the file HTTP needs to change to be changed to utilize Wi-Fi. I read this from a collaborator and I will try to find it and let you know.

**Comment by bdokimakis** [comment]
Same here.

**Comment by CircuitSetup** [comment]
Please try using this build of EmonESP: https://github.com/CircuitSetup/EmonESP

**Comment by bdokimakis** [comment]
Compiling with Arduino IDE 2.3.6 gives the following error:

In file included from C:\Users\bdoki\Downloads\EmonESP-master\EmonESP-master\src\emonesp.h:31,
                 from C:\Users\bdoki\Downloads\EmonESP-master\EmonESP-master\src\src.ino:30:
C:\Users\bdoki\Downloads\EmonESP-master\EmonESP-master\src\debug.h:7:10: fatal error: MicroDebug.h: No such file or directory
    7 | #include "MicroDebug.h"
      |          ^~~~~~~~~~~~~~
compilation terminated.
exit status 1

Compilation error: MicroDebug.h: No such file or directory

**Comment by CircuitSetup** [comment]
Yes, that is a library that needs to be installed, along with others listed in the platformio file

**Comment by bdokimakis** [comment]
I've installed all the libraries that are mentioned in https://github.com/CircuitSetup/EmonESP/blob/master/platformio.ini and could be found via the Library Manager of the Arduino IDE. Micro Debug@^0.0.4 was not among them. Only this https://github.com/rlogiacco/MicroDebug is available and doesn't resolve the problem I mentioned above.

Could you provide a thorough installation guide for Arduino IDE?

**Comment by CircuitSetup** [comment]
> I've installed all the libraries that are mentioned in https://github.com/CircuitSetup/EmonESP/blob/master/platformio.ini and could be found via the Library Manager of the Arduino IDE. Micro Debug@^0.0.4 was not among them. 

This is the library needed: https://github.com/jeremypoulter/MicroDebug
There are also some others written by Jeremy: 
https://github.com/jeremypoulter/StreamSpy
https://github.com/jeremypoulter/ESPAL
https://github.com/jeremypoulter/ConfigJson
